lte/gateway/c/core/oai/tasks/s1ap/s1ap_mme_handlers.c

```diff
@@ -416,6 +428,44 @@ void clean_stale_enb_state(s1ap_state_t* state,
   OAILOG_DEBUG(LOG_S1AP, "Removed stale eNB and all associated UEs.");
 }
 
+static status_code_e s1ap_clear_ue_ctxt_for_unknown_mme_ue_s1ap_id(
+    s1ap_state_t* state, sctp_assoc_id_t sctp_assoc_id) {
+  OAILOG_FUNC_IN(LOG_S1AP);
+  unsigned int i = 0;
+  unsigned int num_elements = 0;
+  bool unlock = true;
+  hash_table_ts_t* hashtblP = get_s1ap_ue_state();
+  hash_node_t *node = NULL, *oldnode = NULL;
+  if (!hashtblP) {
+    OAILOG_ERROR(LOG_S1AP, "No UEs found in comp_s1ap_id hash list");
+    OAILOG_FUNC_RETURN(LOG_S1AP, RETURNerror);
+  }
+  while ((num_elements < hashtblP->num_elements) && (i < hashtblP->size)) {
+    pthread_mutex_lock(&hashtblP->lock_nodes[i]);
+    if (hashtblP->nodes[i] != NULL) {
+      node = hashtblP->nodes[i];
+    }
+    unlock = true;
+    while (node) {
+      num_elements++;
+      oldnode = node;
+      node = node->next;
+      if (oldnode->data &&
+          (sctp_assoc_id ==
+           ((ue_description_t*)oldnode->data)->sctp_assoc_id)) {
+        unlock = false;
+        pthread_mutex_unlock(&hashtblP->lock_nodes[i]);
+        s1ap_remove_ue(state, oldnode->data);
```

Comment:
You need to re-acquire the lock after `s1ap_remove_ue`, since you are still traversing through the list of nodes at `hashtblP->nodes[i]` inside this while loop